### PR TITLE
fix: use websocket transport instead polling

### PR DESCRIPTION
**Pull request by jeonghyeon-net**
I fixed this [Issue](https://github.com/ignoxx/SocketIO-GMS2-Extension/issues/11)

### CODE REVIEW

File: Source/gms2.api.js [hunk 1]
```diff
@@ -12,11 +12,11 @@ class SocketIO {
 
     connect() {
         this.socket = io();
-        this.init_socket_event();
+        this.init_socket_event({ transports: ["websocket", "polling"] });
```
Comment (ignoxx) [inline Source/gms2.api.js]: You're passing the object to the wrong function, you should pass it to io()

Comment (ignoxx) [reply Source/gms2.api.js]: What about creating "this.options = { transports: ["websocket", "polling"] }" in the constructor and reusing it in connect() and connect_to_url()?

Comment (ignoxx) [reply Source/gms2.api.js]: And thanks for the PR! :)

Comment (jeonghyeon-net) [reply Source/gms2.api.js]: > You're passing the object to the wrong function, you should pass it to io()

[Commit](https://github.com/ignoxx/SocketIO-GMS2-Extension/pull/12/commits/c53b95a35ec2c14d0a45c3acc59ccfe1e0c77acb) → Oh, sorry I fixed. Thanks

Comment (jeonghyeon-net) [reply Source/gms2.api.js]: > What about creating "this.options = { transports: ["websocket", "polling"] }" in the constructor and reusing it in connect() and connect_to_url()?

## Why set this?

Currently most browsers support websocket well.  
Initially connecting with websocket will improve performance, improve connection status and save resources.

default is `{ transports: ["polling", "websockets"] }`

> https://socket.io/docs/v3/client-initialization/#transports

In my case, I had to modify transports to connect to the server.

## Reusing

I don't know how to set this in global

Comment (ignoxx) [reply Source/gms2.api.js]: Yea I got the reason, it makes sense to set it to use websockets first. What I had in mind was to have the transport options `{ transports: ["websocket", "polling"] }` defined once in the constructor like 

`this.options =  { transports: ["websocket", "polling"] }` 

and than being re-used in `connect()` and `connect_by_url()` by 

`io(this.options)` `io(url, this.options)`. 

So we don't repeat ourselfs and be less error prone :)

Comment (jeonghyeon-net) [reply Source/gms2.api.js]: > What I had in mind was to have the transport options { transports: ["websocket", "polling"] } defined once in the constructor like

[Commit](https://github.com/ignoxx/SocketIO-GMS2-Extension/pull/12/commits/659ade58761d4ad02c0de4a10cbcd7c41a5a3888) → OK, I Fixed it

Comment (ignoxx) [reply Source/gms2.api.js]: Awesome, thanks!

**Pull request merged**